### ISSUE TITLE
Feature Request: HTTP Proxy Support in Yggdrasil

### ISSUE THREAD

**Issue by lespromtehkms** [yggdrasil-network/yggdrasil-go#1154]
Currently, Yggdrasil lacks built-in support for HTTP proxies. This makes it impossible to use Yggdrasil in environments where all network traffic must be routed through a proxy server.

Currently, it is possible to use tun2socks to route Yggdrasil traffic through an HTTP proxy. However, this solution is not optimal as it complicates configuration and may not be applicable in all cases.

Please consider adding support for HTTP proxies. This would greatly improve the usability of Yggdrasil in environments where proxy usage is mandatory. 

Over *tun2socks* http-proxy:
![image](https://github.com/user-attachments/assets/a1c0499a-5989-41b2-b1ff-bb9336131563)

*Please note that English is not my native language, and I am using a translator to communicate.*

**Comment by basilgello** [comment]
Author of recently added WebSocket PR here. HTTP proxy support is doable but it will work only with WebSocket transports. For all other peer types, SOCKS is needed. Once new Yggdrasil version is officially released, public peers using WebSocket will start to appear.

**Comment by lespromtehkms** [comment]
> Author of recently added WebSocket PR here. HTTP proxy support is doable but it will work only with WebSocket transports. For all other peer types, SOCKS is needed. Once new Yggdrasil version is officially released, public peers using WebSocket will start to appear.

WebSocket doesn't solve the problem of connecting from behind a corporate proxy. Even with peers using ws://ip:port, how do you bypass the proxy? SOCKS allows for that, so it would be desirable to have something like `http://[username]:[password]@[proxyhost]:[proxyport]/[host]:[port]` for that purpose.

**Comment by neilalexander** [comment]
The WebSocket dialer should obey the `HTTP_PROXY` environment variable FWIW. If it doesn't then that's a bug we can fix. Please test from the `develop` branch and let me know.

**Comment by neilalexander** [comment]
Yep, I did a quick test with `HTTP_PROXY=localhost:65432 go run ./cmd/yggdrasil -useconffile ...` and observed that the WebSocket peer attempts to connect to the supplied HTTP proxy.

**Comment by lespromtehkms** [comment]
I'm not a developer, so "develop" is a bit of a black box for me. I used to use Yggdrasil without any issues, but then our head office decided to route all company traffic through a proxy server. This completely messed up my network connectivity, both on my work computer and phone. I had to find a workaround, and tun2socks did the trick. Which got me thinking: since it works that way, why not add native HTTP proxy support to Yggdrasil?

If WebSocket can be configured to work with a proxy by simply using an environment variable, that would be awesome!
